crates/ibc/src/core/ics04_channel/events/channel_attributes.rs

```diff
@@ -0,0 +1,109 @@
+///! This module holds all the abci event attributes for IBC events emitted
+///! during the channel handshake.
+use alloc::string::ToString;
+use derive_more::From;
+use tendermint::abci::tag::Tag;
+
+use crate::core::{
+    ics04_channel::Version,
+    ics24_host::identifier::{ChannelId, ConnectionId, PortId},
+};
+
+const CONNECTION_ID_ATTRIBUTE_KEY: &str = "connection_id";
+const CHANNEL_ID_ATTRIBUTE_KEY: &str = "channel_id";
+const PORT_ID_ATTRIBUTE_KEY: &str = "port_id";
+/// This attribute key is public so that OpenInit can use it to convert itself
+/// to an `AbciEvent`
+pub const COUNTERPARTY_CHANNEL_ID_ATTRIBUTE_KEY: &str = "counterparty_channel_id";
+const COUNTERPARTY_PORT_ID_ATTRIBUTE_KEY: &str = "counterparty_port_id";
+const VERSION_ATTRIBUTE_KEY: &str = "version";
+
+#[derive(Debug, From)]
+pub struct PortIdAttribute {
```

Comment:
Nit: would be nice to have a proc macro attr for such attributes (in the future) -> 
```rust
#[abci_tag(PORT_ID_ATTRIBUTE_KEY)]
pub struct PortIdAttribute { /* ... */ }
```

Reply:
should we track this in an issue?

Reply:
https://github.com/cosmos/ibc-rs/issues/214